src/GafferScene/SetAlgo.cpp

```diff
@@ -308,6 +318,11 @@ struct AstHasher
 		boost::apply_visitor( *this, expr.right.expr );
 	}
 
+	void operator()( const Nil &nil )
+	{
+		m_hash.append( false );
```

Comment:
One of the tests in `SetFilterTest` runs the node with an empty string as the set name/expression. I had to explicitly hash a dummy value here, so that Gaffer doesn't throw an exception at me because `SetFilter::hash()` never actually hashes anything in that case. Do you have a better idea?